Clean up ESLint errors

(continued from #39 conversation)

```sh
$$ npm run lint

> mozilla-universal-search-addon@1.0.1 lint /Users/pdehaan/dev/github/universal-search-addon
> eslint .


bootstrap.js
   5:17  error  Cc is defined but never used         no-unused-vars
   5:33  error  Ci is defined but never used         no-unused-vars
   5:57  error  Cm is defined but never used         no-unused-vars
  11:9   error  startup is defined but never used    no-unused-vars
  16:9   error  shutdown is defined but never used   no-unused-vars
  32:9   error  install is defined but never used    no-unused-vars
  38:9   error  uninstall is defined but never used  no-unused-vars

lib/main.js
    7:17  error  Cc is defined but never used                no-unused-vars
    7:33  error  Ci is defined but never used                no-unused-vars
    7:57  error  Cm is defined but never used                no-unused-vars
   18:4   error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars
   86:6   error  document is defined but never used          no-unused-vars
  143:4   error  Main is defined but never used              no-unused-vars

lib/ui/GoButton.js
  5:4  error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars

lib/ui/Popup.js
   14:4  error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars
  257:4  error  Expected no return value                    consistent-return

lib/ui/Urlbar.js
  7:4  error  EXPORTED_SYMBOLS is defined but never used  no-unused-vars

✖ 17 problems (17 errors, 0 warnings)
```

I can probably clean up most of these by just ignoring the `no-unused-vars` rule on specific lines without needing to disable the rule globally.

I'll try and come up with a better explanation of the `consistent-return` issue while I'm at it.

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,0 +1,5 @@
+extends: universal-search
+
+rules:
+  consistent-return: 0
+  no-console: 0

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 node_modules
 dist
+*.log

--- a/bootstrap.js
+++ b/bootstrap.js
@@ -1,22 +1,24 @@
 'use strict';
 
-const { classes: Cc, interfaces: Ci, utils: Cu, manager: Cm } = Components;
+/* global APP_SHUTDOWN, ADDON_DISABLE, ADDON_UNINSTALL, Components, XPCOMUtils, Main */
+
+const { utils: Cu } = Components;
 
 Cu.import('resource://gre/modules/XPCOMUtils.jsm');
 XPCOMUtils.defineLazyModuleGetter(this, 'Main',
   'chrome://universalsearch-lib/content/main.js');
 
-function startup(data, reason) {
+function startup(data, reason) { // eslint-disable-line no-unused-vars
   console.log('bootstrap startup called');
   Main.load();
 }
 
-function shutdown(data, reason) {
+function shutdown(data, reason) { // eslint-disable-line no-unused-vars
   console.log('bootstrap shutdown called');
   // no teardown is needed for a normal shutdown
-  if (reason == APP_SHUTDOWN) { return; }
+  if (reason === APP_SHUTDOWN) { return; }
 
-  if (reason == ADDON_DISABLE || reason == ADDON_UNINSTALL) {
+  if (reason === ADDON_DISABLE || reason === ADDON_UNINSTALL) {
     // uninstall / offboarding experience? ask for feedback?
     // TODO: xhr the uninstall event to a server
   }
@@ -27,12 +29,12 @@ function shutdown(data, reason) {
   Main.unload();
 }
 
-function install(data, reason) {
+function install(data, reason) { // eslint-disable-line no-unused-vars
   console.log('bootstrap install called');
   // upsell? product tour? other first time experience?
   // TODO: xhr the install event to a server
 }
 
-function uninstall(data, reason) {
+function uninstall(data, reason) { // eslint-disable-line no-unused-vars
   console.log('bootstrap uninstall called');
 }

--- a/lib/main.js
+++ b/lib/main.js
@@ -2,7 +2,10 @@
 //       figure out when and how to cache-bust.
 //       bugs 918033, 1051238, 719376
 
-const { classes: Cc, interfaces: Ci, utils: Cu, manager: Cm } = Components;
+/* global Components, CustomizableUI, Services, XPCOMUtils */
+
+const { utils: Cu } = Components;
+
 Cu.import('resource://gre/modules/XPCOMUtils.jsm');
 XPCOMUtils.defineLazyModuleGetter(this, 'Services',
   'resource://gre/modules/Services.jsm');
@@ -13,7 +16,7 @@ XPCOMUtils.defineLazyModuleGetter(this, 'console',
 XPCOMUtils.defineLazyModuleGetter(this, 'CustomizableUI',
   'resource:///modules/CustomizableUI.jsm');
 
-var EXPORTED_SYMBOLS = ['Main'];
+var EXPORTED_SYMBOLS = ['Main']; // eslint-disable-line no-unused-vars
 
 var onTabSelect = function() { console.log('onTabSelect'); };
 var onTabOpen = function() { console.log('onTabOpen'); };
@@ -25,10 +28,10 @@ var loadIntoWindow = function(win) {
   var document = win.document;
 
   // set the app global per-window
-  if(win.US === undefined) {
-      Object.defineProperty(win, 'US', {configurable:true, value:{}});
+  if (win.US === undefined) {
+    Object.defineProperty(win, 'US', {configurable: true, value: {}});
   } else {
-      win.US = win.US || {};
+    win.US = win.US || {};
   }
 
   // hide the search bar, if it's visible; this will be null if not
@@ -74,14 +77,13 @@ var loadIntoWindow = function(win) {
   // browser element.
   win.US.setBrowser = function(browserEl) {
     win.US.browser = browserEl;
-  }
+  };
 };
 
 // basically reverse the loadIntoWindow function
 var unloadFromWindow = function(win) {
   console.log('unloadFromWindow start');
 
-  var document = win.document;
   win.US.goButton.derender(win);
   win.gBrowser.tabContainer.removeEventListener('TabSelect', onTabSelect);
   win.gBrowser.tabContainer.removeEventListener('TabOpen', onTabOpen);
@@ -105,7 +107,7 @@ function onWindowNotification(win, topic) {
   console.log('iterating windows');
   win.addEventListener('load', function loader() {
     win.removeEventListener('load', loader, false);
-    if (win.location.href == 'chrome://browser/content/browser.xul') {
+    if (win.location.href === 'chrome://browser/content/browser.xul') {
       loadIntoWindow(win);
     }
   }, false);
@@ -116,14 +118,14 @@ function load() {
   var enumerator = Services.wm.getEnumerator('navigator:browser');
   while (enumerator.hasMoreElements()) {
     var win = enumerator.getNext();
-    try { 
+    try {
       loadIntoWindow(win);
     } catch (ex) {
       console.log('load into window failed: ', ex);
     }
   }
   Services.ww.registerNotification(onWindowNotification);
-};
+}
 
 function unload() {
   var enumerator = Services.wm.getEnumerator('navigator:browser');
@@ -138,4 +140,4 @@ function unload() {
   Services.ww.unregisterNotification(onWindowNotification);
 }
 
-var Main = { load: load, unload: unload };
+var Main = { load: load, unload: unload }; // eslint-disable-line no-unused-vars

--- a/lib/ui/GoButton.js
+++ b/lib/ui/GoButton.js
@@ -2,7 +2,7 @@
 
 'use strict';
 
-var EXPORTED_SYMBOLS = [ 'GoButton' ];
+var EXPORTED_SYMBOLS = [ 'GoButton' ]; // eslint-disable-line no-unused-vars
 
 function GoButton() {}
 GoButton.prototype = {
@@ -22,4 +22,3 @@ GoButton.prototype = {
     return false; // let's not fall through for now
   }
 };
-

--- a/lib/ui/Popup.js
+++ b/lib/ui/Popup.js
@@ -2,6 +2,8 @@
 
 'use strict';
 
+/* global Cc, Ci, Cu, Components, SearchSuggestionController, Services, XPCOMUtils, WebChannel */
+
 XPCOMUtils.defineLazyModuleGetter(this, 'SearchSuggestionController',
   'resource://gre/modules/SearchSuggestionController.jsm');
 XPCOMUtils.defineLazyModuleGetter(this, 'Promise',
@@ -9,10 +11,10 @@ XPCOMUtils.defineLazyModuleGetter(this, 'Promise',
 XPCOMUtils.defineLazyModuleGetter(this, 'WebChannel',
   'resource://gre/modules/WebChannel.jsm');
 
-var EXPORTED_SYMBOLS = [ 'Popup' ];
+var EXPORTED_SYMBOLS = [ 'Popup' ]; // eslint-disable-line no-unused-vars
 
 function Popup() {
-  var prefBranch = Cc["@mozilla.org/preferences-service;1"].getService(Ci.nsIPrefService).getBranch("");
+  var prefBranch = Cc['@mozilla.org/preferences-service;1'].getService(Ci.nsIPrefService).getBranch('');
   this.channelID = 'ohai';
   this.frameURL = prefBranch.getPrefType('services.universalSearch.frameURL') ?
                    prefBranch.getCharPref('services.universalSearch.frameURL') :
@@ -20,14 +22,15 @@ function Popup() {
   this.frameBaseURL = prefBranch.getPrefType('services.universalSearch.baseURL') ?
                        prefBranch.getCharPref('services.universalSearch.baseURL') :
                        'https://d1fnkpeapwua2i.cloudfront.net';
-};
+}
+
 Popup.prototype = {
   constructor: Popup,
   port: null,
   browser: null,
   // setting isPinned to true will force the popup to stay open forever
   isPinned: false,
-  render:  function(win) {
+  render: function(win) {
     this.popup = win.document.createElementNS('http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul', 'panel');
     this.popup.setAttribute('type', 'autocomplete-richlistbox');
     this.popup.setAttribute('id', 'PopupAutoCompleteRichResultUnivSearch');
@@ -79,11 +82,11 @@ Popup.prototype = {
   onMessage: function(id, msg, sender) {
     var engine = Services.search.defaultEngine;
     var url;
-    if (id != this.channelID) { return; }
-    if (msg.type == 'autocomplete-url-clicked') {
+    if (id !== this.channelID) { return; }
+    if (msg.type === 'autocomplete-url-clicked') {
       console.log('browser received "autocomplete-url-clicked" event: ', msg);
       this.popup.hidePopup();
-      if (msg.data.resultType == 'url') {
+      if (msg.data.resultType === 'url') {
         // it's navigable, go for it
         window.US.gURLBar.inputField.value = msg.data.result;
         window.gBrowser.loadURI(msg.data.result);
@@ -94,9 +97,9 @@ Popup.prototype = {
         window.US.gURLBar.inputField.value = url;
         window.gBrowser.loadURI(url);
       }
-    } else if (msg.type == 'url-selected') {
+    } else if (msg.type === 'url-selected') {
       console.log('browser received "url-selected" event: ', msg);
-      if (msg.data.resultType == 'url') {
+      if (msg.data.resultType === 'url') {
         window.US.gURLBar.value = msg.data.result;
         window.US.gURLBar._search = null;
       } else {
@@ -119,7 +122,7 @@ Popup.prototype = {
     console.log('browser sending message to iframe: ', msg);
     this.port.send(msg, {
       browser: this.browser,
-      principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+      principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
     });
   },
   onPopupHiding: function(evt) {
@@ -129,17 +132,17 @@ Popup.prototype = {
     console.log('browser sending message to iframe: ', msg);
     this.port.send(msg, {
       browser: this.browser,
-      principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+      principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
     });
   },
   _getImageURLForResolution: function(aWin, aURL, aWidth, aHeight) {
     if (!aURL.endsWith('.ico') && !aURL.endsWith('.ICO')) {
       return aURL;
     }
-    let width  = Math.round(aWidth * aWin.devicePixelRatio);
+    let width = Math.round(aWidth * aWin.devicePixelRatio);
     let height = Math.round(aHeight * aWin.devicePixelRatio);
-    return aURL + (aURL.contains("#") ? "&" : "#") +
-           "-moz-resolution=" + width + "," + height;
+    return aURL + (aURL.contains('#') ? '&' : '#') +
+           '-moz-resolution=' + width + ',' + height;
 
   },
   _appendCurrentResult: function() {
@@ -159,7 +162,7 @@ Popup.prototype = {
       console.log('browser sending message to iframe: ', autocompleteMsg);
       this.port.send(autocompleteMsg, {
         browser: this.browser,
-        principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+        principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
       });
 
       delete searchSuggestions.formHistoryResult;
@@ -167,19 +170,19 @@ Popup.prototype = {
       console.log('browser sending message to iframe: ', suggestedMsg);
       this.port.send(suggestedMsg, {
         browser: this.browser,
-        principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+        principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
       });
     }.bind(this), function(err) {
       Cu.reportError(err);
       console.log('browser sending message to iframe: ', autocompleteMsg);
       this.port.send(autocompleteMsg, {
         browser: this.browser,
-        principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+        principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
       });
       console.log('browser sending message to iframe: ', suggestedMsg);
       this.port.send(suggestedMsg, {
         browser: this.browser,
-        principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+        principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
       });
     });
   },
@@ -199,13 +202,13 @@ Popup.prototype = {
         var imgMatches = chromeImgLink.match(/^moz-anno\:favicon\:(.*)/);
 
         results.push({
-          url: Components.classes["@mozilla.org/intl/texttosuburi;1"].
+          url: Components.classes['@mozilla.org/intl/texttosuburi;1'].
                 getService(Components.interfaces.nsITextToSubURI).
-                unEscapeURIForUI("UTF-8", controller.getValueAt(i)),
+                unEscapeURIForUI('UTF-8', controller.getValueAt(i)),
           image: imgMatches ? imgMatches[1] : null,
           title: controller.getCommentAt(i),
           type: controller.getStyleAt(i),
-          text: controller.searchString.replace(/^\s+/, "").replace(/\s+$/, "")
+          text: controller.searchString.trim()
         });
       }
     }
@@ -221,7 +224,7 @@ Popup.prototype = {
     // figure out which message manager to pass into ContentSearch, in order to get the response message back.
     // it's possible all of this code was unnecessary and we could just fire a GetSuggestions message into
     // the ether, and fully expect to get a Suggestions object back with the suggestions. /me shrugs
-    // 
+    //
     //var suggestionData = { engineName: engine.name, searchString: gURLBar.inputField.value, remoteTimeout: 5000 };
     //ContentSearch._onMessageGetSuggestions(brow.messageManager, suggestionData);
     var controller = this.popup.mInput.controller;
@@ -237,7 +240,7 @@ Popup.prototype = {
 
     // searchTerm is the same thing as the 'text' item sent down in each result.
     // maybe that's not a useful place to put the search term...
-    let searchTerm = controller.searchString.replace(/^\s+/, "").replace(/\s+$/, "")
+    let searchTerm = controller.searchString.trim();
 
     // unfortunately, the controller wants to do some UI twiddling.
     // and we don't have any UI to give it. so it barfs.

--- a/lib/ui/Urlbar.js
+++ b/lib/ui/Urlbar.js
@@ -2,7 +2,9 @@
 
 'use strict';
 
-var EXPORTED_SYMBOLS = ['Urlbar'];
+/* global Cc, Ci, US */
+
+var EXPORTED_SYMBOLS = ['Urlbar']; // eslint-disable-line no-unused-vars
 
 function Urlbar() {}
 Urlbar.prototype = {
@@ -68,11 +70,11 @@ Urlbar.prototype = {
       // TODO: when we reimplement the controller, actually check the location of the cursor...the backspace
       // could do nothing if it was in front of the character. And we shoudl probably check for cursor location
       // and listen for the Del key too. And the trim() call could also mess things up. Selections are a real PITA ;-)
-      if (evt.key == 'Backspace' && gURLBar.inputField.value.trim().length > 1) { return; }
+      if (evt.key === 'Backspace' && gURLBar.inputField.value.trim().length > 1) { return; }
 
       // ok, if the user selected something, we set state on gURLBar.
-      if (evt.key == 'Enter') {
-        if (gURLBar.search && gURLBar.search.term == gURLBar.value) {
+      if (evt.key === 'Enter') {
+        if (gURLBar.search && gURLBar.search.term === gURLBar.value) {
           // the user selected a search suggestion via keypresses, then hit enter.
           // we know the suggestion is not stale, because it matches the current contents
           // of the urlbar. So, navigate to the hidden search URL that corresponds to the term.
@@ -98,10 +100,10 @@ Urlbar.prototype = {
           shiftKey: evt.shiftKey
         }
       };
-      console.log('browser sending message to iframe: ',msg);
+      console.log('browser sending message to iframe: ', msg);
       US.popup.port.send(msg, {
         browser: US.popup.browser,
-        principal: Cc["@mozilla.org/systemprincipal;1"].createInstance(Ci.nsIPrincipal)
+        principal: Cc['@mozilla.org/systemprincipal;1'].createInstance(Ci.nsIPrincipal)
       });
     }
 

--- a/package.json
+++ b/package.json
@@ -1,26 +1,31 @@
 {
   "name": "mozilla-universal-search-addon",
-  "version": "1.1.0",
   "description": "universal search desktop experiments in addon format",
-  "main": "index.js",
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/mozilla/universal-search-addon"
-  },
-  "author": "",
-  "license": "MPL2",
+  "version": "1.1.0",
+  "author": "Mozilla (https://mozilla.org/)",
   "bugs": {
     "url": "https://github.com/mozilla/universal-search-addon/issues"
   },
-  "homepage": "https://github.com/mozilla/universal-search-addon",
   "dependencies": {
     "jpm": "^1.0.0"
+  },
+  "devDependencies": {
+    "eslint": "0.24.0",
+    "eslint-config-universal-search": "1.0.0"
   },
   "engines": {
     "firefox": ">=38.0a1",
     "fennec": ">=38.0a1"
+  },
+  "homepage": "https://github.com/mozilla/universal-search-addon",
+  "license": "MPL-2.0",
+  "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/mozilla/universal-search-addon"
+  },
+  "scripts": {
+    "lint": "eslint .",
+    "test": "echo \"Error: no test specified\" && exit 1"
   }
 }

--- a/test/test-index.js
+++ b/test/test-index.js
@@ -1,19 +1,19 @@
-var main = require("../");
+var main = require('../');
 
-exports["test main"] = function(assert) {
-  assert.pass("Unit test running!");
+exports['test main'] = function(assert) {
+  assert.pass('Unit test running!');
 };
 
-exports["test main async"] = function(assert, done) {
-  assert.pass("async Unit test running!");
+exports['test main async'] = function(assert, done) {
+  assert.pass('async Unit test running!');
   done();
 };
 
-exports["test dummy"] = function(assert, done) {
-  main.dummy("foo", function(text) {
-    assert.ok((text === "foo"), "Is the text actually 'foo'");
+exports['test dummy'] = function(assert, done) {
+  main.dummy('foo', function(text) {
+    assert.ok((text === 'foo'), 'Is the text actually \'foo\'');
     done();
   });
 };
 
-require("sdk/test").run(exports);
+require('sdk/test').run(exports);

